backend/templates/users/friend_list.html

```diff
@@ -0,0 +1,80 @@
+{% extends "base.html" %}
+{% load static %}
+
+{% block title %}Friends List{% endblock %}
+
+{% block content %}
+<h1>Friends List</h1>
+
+<div class="main border">
+    <h2>Online Friends</h2>
+    <ul>
+        {% if online_friends %}
+        {% for friend in online_friends %}
+        <li>{{ friend.username }} (Online)</li>
+        {% endfor %}
+        {% else %}
+        <li>No online friends.</li>
+        {% endif %}
+    </ul>
+</div>
+<div class="main border">
+    <h2>All Friends</h2>
+    <ul>
+        {% if friends %}
+        {% for friend in friends %}
+        <li>{{ friend.username }} {% if friend in online_friends %}(Online){% else %}(Offline){% endif %}</li>
+        {% endfor %}
+        {% else %}
+        <li>No friends.</li>
+        {% endif %}
+    </ul>
+</div>
+
+<div class="main border">
+    <h2>Pending Friend Requests</h2>
+    <ul>
+        {% if pending_requests %}
+        {% for request in pending_requests %}
+        <li>
+            {{ request.username }} wants to be your friend.
+             <a href="{% url 'accept_friend' request.username %}">Accept</a> |
+            <a href="{% url 'reject_friend' request.username %}">Reject</a>
+        </li>
+        {% endfor %}
+        {% else %}
+        <li>No pending friend requests.</li>
+        {% endif %}
+    </ul>
+</div>
+
+<!-- JavaScript to handle the modal -->
+<script>
+    document.addEventListener('DOMContentLoaded', function() {
+        const errorMessage = "{{ error_message|escapejs }}";
+        if (errorMessage) {
+            document.getElementById('errorMessage').innerText = errorMessage;
+            const errorModal = new bootstrap.Modal(document.getElementById('errorModal'));
+            errorModal.show();
+        }
+    });
+</script>
+<!-- Modal for error messages -->
+<div class="modal fade" id="errorModal" tabindex="-1" aria-labelledby="errorModalLabel" aria-hidden="true" data-bs-theme="dark">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h5 class="modal-title" id="errorModalLabel">Error</h5>
+                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+            </div>
+            <div class="modal-body">
+                <p id="errorMessage" ></p>
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
+            </div>
+        </div>
+    </div>
+</div>
```

Comment:
I am trying to avoid using js functionalities in the front end and rather use the backend to handle error and allow functionality, specially that I think it is cleaner this way. But we can go to this later after we are sure that it works well.